contrib/opencensus-ext-requests/opencensus/ext/requests/trace.py

```diff
@@ -74,22 +79,53 @@ def call(url, *args, **kwargs):
         if utils.disable_tracing_hostname(dest_url, blacklist_hostnames):
             return requests_func(url, *args, **kwargs)
 
+        path = parsed_url.path if parsed_url.path else '/'
+
         _tracer = execution_context.get_opencensus_tracer()
         _span = _tracer.start_span()
-        _span.name = '[requests]{}'.format(requests_func.__name__)
+        _span.name = '{}'.format(path)
         _span.span_kind = span_module.SpanKind.CLIENT
 
-        # Add the requests url to attributes
-        _tracer.add_attribute_to_current_span(HTTP_URL, url)
+        # Add the requests host to attributes
+        _tracer.add_attribute_to_current_span(
+            HTTP_HOST, dest_url)
 
-        result = requests_func(url, *args, **kwargs)
+        # Add the requests method to attributes
+        _tracer.add_attribute_to_current_span(
+            HTTP_METHOD, requests_func.__name__.upper())
 
-        # Add the status code to attributes
+        # Add the requests path to attributes
         _tracer.add_attribute_to_current_span(
-            HTTP_STATUS_CODE, str(result.status_code))
+            HTTP_PATH, path)
 
-        _tracer.end_span()
-        return result
+        # Add the requests url to attributes
+        _tracer.add_attribute_to_current_span(HTTP_URL, url)
+
+        try:
+            result = requests_func(url, *args, **kwargs)
+        except requests.Timeout:
+            _span.set_status(
+                status_module.Status(1, message='request timed out')
+            )
+        except requests.URLRequired:
+            _span.set_status(
+                status_module.Status(3, message='invalid URL')
+            )
+        except Exception as e:
+            _span.set_status(
+                status_module.Status(2, message=str(e))
+            )
+        else:
+            # Add the status code to attributes
+            _tracer.add_attribute_to_current_span(
+                HTTP_STATUS_CODE, str(result.status_code)
```

Comment:
According to the spec, `status_code` is an int64. https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#mapping-from-http-status-codes-to-trace-status-codes

This again seems to be a spec bug in the [test-cases](https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#test-cases) section, where `status_code` is put as a string.

Reply:
I think it is better to leave as int64, following specs. But open for discussion!

Reply:
> According to the spec, `status_code` is an int64. https://github.com/census-instrumentation/opencensus-specs/blob/master/trace/HTTP.md#mapping-from-http-status-codes-to-trace-status-codes

Right now `AzureExporter` for tracer is broken because `Flask`, `Django` and `Pyramid` integrations set `status_code` as string, and azures expect int, we can typecast on azure or change the other integrations, which way is preferred?

Reply:
https://github.com/census-instrumentation/opencensus-python/blame/master/contrib/opencensus-ext-azure/opencensus/ext/azure/trace_exporter/__init__.py#L86

My understanding is that we'll still get the right `status_code`, but the `success` flag will be wrong.

Reply:
> My understanding is that we'll still get the right `status_code`, but the `success` flag will be wrong.

Yes, and with this we have both options described above. I can take any of them, I prefer to put the `status_code` as int, it makes more sense to me.

Reply:
I agree with you that we should use int. Converting int to string and then parse it to int sounds like crazy :)

Reply:
I will open a new PR to fix this into `Flask`, `Django` and `Pyramid` integrations.

Reply:
Did it [here](https://github.com/census-instrumentation/opencensus-python/pull/755).